src/main/java/hudson/slaves/DummyCloudImpl.java

```diff
@@ -123,15 +121,8 @@ public Node call() throws Exception {
             
             System.out.println("launching slave");
             final DumbSlave slave = rule.createSlave(label);
-            Provider<NodeProperty> nodePropertyProvider = new Provider<NodeProperty>() {
-                @Override
-                public NodeProperty invoke(NodeProperty request) {
-                    request.setNode(slave);
-                    return request;
-                }
-            };
             for (NodeProperty nodeProperty : nodeProperties) {
-                slave.getNodeProperties().add(nodePropertyProvider.invoke(nodeProperty));
+                slave.getNodeProperties().add(updateWithNode(nodeProperty, slave));
```

Comment:
nit: Unless I'm missing something it looks like you could just inline the method instead:
    
    nodeProperty.setNode(slave);
    slave.getNodeProperties().add(nodeProperty);

Reply:
Yeah :( 
Whatever, I doubt we will touch this class till JDK 15 or so

Reply:
Yep definitely not important, just figured I'd mention it :)

Reply:
I cannot figure out what the original code here was doing. Introduced in https://github.com/jenkinsci/jenkins/pull/1815 FTR.